dask/array/reductions.py

```diff
@@ -1583,7 +1583,16 @@ def median(a, axis=None, keepdims=False, out=None):
 
     axis = [ax + a.ndim if ax < 0 else ax for ax in axis]
 
-    a = a.rechunk({ax: -1 if ax in axis else "auto" for ax in range(a.ndim)})
+    # rechunk if reduced axes are not contained in a single chunk
+    # cannot use `any` to check since `any` is defined in this file
+    needs_rechunk = False
+    for ax in axis:
+        if a.numblocks[ax] > 1:
+            needs_rechunk = True
+            break
+
+    if needs_rechunk:
```

Comment:
Maybe this?

```suggestion
    if any(a.numblocks[ax] > 1 for ax in axis):
```

Reply:
I put a comment above, but maybe it's not clear. `any` is overridden in this file.

Reply:
Ah, my apologies for not reading.  We should probably enable devs to use any.  Maybe the following?

```python
import builtins  # at the top of the file
```

```python
if builtins.any(...)
```

Reply:
ooh did not know about that :)